ykman-gui/py/yubikey.py

```diff
@@ -237,20 +196,32 @@ def write_config(self, usb_applications, nfc_applications, lock_code):
             return success()
 
     def refresh_piv(self):
-        with self._open_piv() as piv_controller:
+        with self._open_device() as conn:
+            session = PivSession(conn)
+            pivman = get_pivman_data(session)
+
             return success({
                 'piv_data': {
-                    'certs': self._piv_list_certificates(piv_controller),
-                    'has_derived_key': piv_controller.has_derived_key,
-                    'has_protected_key': piv_controller.has_protected_key,
-                    'has_stored_key': piv_controller.has_stored_key,
-                    'pin_tries': piv_controller.get_pin_tries(),
-                    'puk_blocked': piv_controller.puk_blocked,
-                    'supported_algorithms':
-                        [a.name for a in piv_controller.supported_algorithms],
+                    'certs': self._piv_list_certificates(session),
+                    'has_derived_key': pivman.has_derived_key,
+                    'has_protected_key': pivman.has_protected_key,
+                    'has_stored_key': pivman.has_stored_key,
+                    'pin_tries': session.get_pin_attempts(),
+                    'puk_blocked': pivman.puk_blocked,
+                    'supported_algorithms': self._supported_algorithms(self._dev_info['version'].split('.')),
                 },
             })
 
+    def _supported_algorithms(self, version):
```

Comment:
There are several private methods like this one (and others for PIV) which could be functions outside of the class (just skip the `self` parameter). Not a huge thing, but it simplifies the class a bit.